src/SourceBuild/content/eng/tools/BinaryToolKit/BinaryTool/Driver.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+
+namespace BinaryTool;
+
+public static class Driver
+{
+    // Directory to scan for binaries: required
+    public static string TargetDirectory { get; set; } = string.Empty;
+
+    // Directory to output reports: required
+    public static string OutputReportDirectory { get; set; } = string.Empty;
+
+    // File for the baseline of allowed retainable binaries: optional
+    public static string AllowedBinariesKeepFile { get; set; } = string.Empty;
+
+    // File for the baseline of allowed removeable binaries: optional
+    public static string AllowedBinariesRemoveFile { get; set; } = string.Empty;
+
+    // Output files
+    public static string DetectedBinariesFile { get; private set; } = string.Empty;
+    public static string UpdatedAllowedBinariesKeepFile { get; private set; } = string.Empty;
+    public static string UpdatedAllowedBinariesRemoveFile { get; private set; } = string.Empty;
+    public static string NewBinariesFile { get; private set; } = string.Empty;
+    public static string RemovedBinariesFile { get; private set; } = string.Empty;
+
+    // Logger
+    public static ILogger Log { get; private set; } = ConfigureLogger();
+
+    public static void Execute()
+    {
+        DateTime startTime = DateTime.Now;
+        Log.LogInformation($"Starting binary detection, validation, and removal tool from {Environment.CurrentDirectory}...");
+
+        ProcessParameters();
+        DetectBinaries.Execute();
+        CompareBinariesAgainstBaseline.Execute();
+        RemoveBinaries.Execute();
+
+        Log.LogInformation("Finished binary detection, validation, and removal tool. Took " + (DateTime.Now - startTime).TotalSeconds + " seconds.");
+    }
+
+    private static ILogger ConfigureLogger()
+    {
+        using ILoggerFactory loggerFactory =
+            LoggerFactory.Create(builder =>
+                builder.AddSimpleConsole(options =>
+                {
+                    options.IncludeScopes = false;
+                    options.SingleLine = true;
+                    options.TimestampFormat = "HH:mm:ss ";
```

Comment:
Is the space at the end intentional?

Reply:
Yes, otherwise the message looks like "TimestampLogLevel" instead of "Timestamp LogLevel"